satpy/readers/nwcsaf_hrw_nc.py

```diff
@@ -0,0 +1,331 @@
+from satpy.readers.nwcsaf_nc import PLATFORM_NAMES, SENSOR, read_nwcsaf_time
+from satpy.utils import get_chunk_size_limit
+
+logger = logging.getLogger(__name__)
+
+CHUNK_SIZE = get_chunk_size_limit()
+
+WIND_CHANNELS = [
+    "wind_hrvis",
+    "wind_ir108",
+    "wind_ir120",
+    "wind_vis06",
+    "wind_vis08",
+    "wind_wv062",
+    "wind_wv073",
+]
+
+# Source: NWC/CDOP3/GEO/AEMET/SW/DOF
+DATASET_UNITS = {
+    "air_pressure": "Pa",
+    "air_pressure_correction": "Pa",
+    "air_pressure_error": "Pa",
+    "air_pressure_nwp_at_best_fit_level": "Pa",
+    "air_temperature": "K",
+    "cloud_type": "1",
+    "correlation": "%",
+    "correlation_test": "1",
+    "height_assignment_method": "1",
+    "latitude": "degree_north",
+    "latitude_increment": "degree_north",
+    "longitude": "degree_east",
+    "longitude_increment": "degree_east",
+    "number_of_winds": "1",
+    "orographic_index": "1",
+    "previous_wind_idx": "1",
+    "quality_index_iwwg_value": "%",
+    "quality_index_with_forecast": "%",
+    "quality_index_without_forecast": "%",
+    "quality_test": "1",
+    "segment_x": "m",
+    "segment_x_pix": "1",
+    "segment_y": "m",
+    "segment_y_pix": "1",
+    "tracer_correlation_method": "1",
+    "tracer_type": "1",
+    "wind_from_direction": "degree",
+    "wind_from_direction_difference_nwp_at_amv_level": "degree",
+    "wind_from_direction_difference_nwp_at_best_fit_level": "degree",
+    "wind_from_direction_nwp_at_amv_level": "degree",
+    "wind_from_direction_nwp_at_best_fit_level": "degree",
+    "wind_idx": "1",
+    "wind_speed": "m/s",
+    "wind_speed_difference_nwp_at_amv_level": "m/s",
+    "wind_speed_difference_nwp_at_best_fit_level": "m/s",
+    "wind_speed_nwp_at_amv_level": "m/s",
+    "wind_speed_nwp_at_best_fit_level": "m/s"
+}
+
+
+class NWCSAFGEOHRWFileHandler(BaseFileHandler):
+    """A file handler class for NWC SAF GEO HRW files."""
+
+    def __init__(self, filename, filename_info, filetype_info, merge_channels=False):
+        """Initialize the file handler."""
+        super().__init__(filename, filename_info, filetype_info)
+
+        self.h5f = h5py.File(self.filename, "r")
+        self.filename_info = filename_info
+        self.filetype_info = filetype_info
+        self.merge_channels = merge_channels
+        self.platform_name = PLATFORM_NAMES.get(self.h5f.attrs["satellite_identifier"].astype(str))
+        self.sensor = SENSOR.get(self.platform_name, "seviri")
+        self.lons = {}
+        self.lats = {}
+        # Imaging period, which is set after reading any data, and used to calculate end time
+        self.period = None
+
+        # The resolution is given in kilometers, convert to meters
+        self.resolution = 1000 * self.h5f.attrs["spatial_resolution"].item()
+
+    def __del__(self):
+        """Close file handlers when we are done."""
+        with suppress(OSError):
+            self.h5f.close()
+
+    def available_datasets(self, configured_datasets=None):
```

Comment:
The `configured_datasets` are not "forwarded on" as suggested in the base file handler:

https://github.com/pytroll/satpy/blob/f9150741f984f6784436be8cda8ea82c23b1c2c2/satpy/readers/file_handlers.py#L256-L263

Without this users will not be able to statically define datasets in the YAML.